client/my-sites/plugins/plugins-browser/index.jsx

```diff
@@ -287,7 +288,18 @@ const PluginsBrowser = React.createClass( {
 		);
 	},
 
-	getManageButton() {
+	shouldShowManageButton() {
+		if ( this.props.isJetpackSite ) {
+			return true;
+		}
+		return ( ! this.props.selectedSiteId && this.props.hasJetpackSites );
```

Comment:
Loading with empty redux state on a simple site, I get a flash of the manage button. Possible that the logic could be tweaked here to avoid that.

Reply:
I have trouble reproducing this and I don't immediately see any flaw in the logic. Could you record a video? Maybe that will give me a clue.

I'll merge this PR now and will solve any remaining problems separately.

Reply:
Here it is:

![manage_flash](https://user-images.githubusercontent.com/7767559/30183317-e1ab8234-9411-11e7-8a14-3948e5b04edb.gif)

`hasJetpackSites` gets populated before `selectedSiteId`. I'm not too sure how this would be fixed. Possible that  `getSelectedSiteId()` should distinguish between _no site_ selected and _not yet known_. Currently it returns `null` for both.

Reply:
Seems way beyond the scope of current work, anyway.